### PR TITLE
Unit test and settings.xml to use Google repository

### LINKED ISSUE #732: Use Google's Maven repository to test
> if there's a google repository we could point to for testConfigureAdditionalMavenRepositories_addingSpringRepository instead of Spring? E.g. https://dl.google.com/dl/android/maven2/index.html

via https://github.com/GoogleCloudPlatform/cloud-opensource-java/pull/729#issuecomment-503119687

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #732 

> With a simple change to your settings.xml configuration file, a build system running on Cloud Platform – for example, Jenkins on Google Compute Engine or Google Cloud Shell – can now fetch your project’s dependencies from Cloud Storage, increasing the speed of your builds.

https://cloudplatform.googleblog.com/2015/11/faster-builds-for-Java-developers-with-Maven-Central-mirror.html

Kokoro output shows it now uses the GCS-backed Maven mirror:

```
T:\src\github\cloud-opensource-java>mvn -s settings.xml -B clean install javadoc:jar
[INFO] Scanning for projects...
[INFO] Downloading: https://maven-central.storage-download.googleapis.com/repos/central/data/com/google/guava/guava-bom/28.0-android/guava-bom-28.0-android.pom
[INFO] Downloaded: https://maven-central.storage-download.googleapis.com/repos/central/data/com/google/guava/guava-bom/28.0-android/guava-bom-28.0-android.pom (2 KB at 1.6 KB/sec)
[INFO] Downloading: https://maven-central.storage-download.googleapis.com/repos/central/data/org/sonatype/oss/oss-parent/9/oss-parent-9.pom
[INFO] Downloaded: https://maven-central.storage-download.googleapis.com/repos/central/data/org/sonatype/oss/oss-parent/9/oss-parent-9.pom (7 KB at 51.3 KB/sec)
```

### CODE REVIEW

**Comment by suztomo** [comment]
Maven central issue? But it's always on kokoro-windows.

```
[ERROR] testFindUpdates_beam(com.google.cloud.tools.opensource.dependencies.DependencyGraphIntegrationTest)  Time elapsed: 68.843 s  <<< ERROR!
org.eclipse.aether.collection.DependencyCollectionException: Failed to collect dependencies at org.apache.beam:beam-sdks-java-io-google-cloud-platform:jar:2.5.0 -> com.google.api.grpc:grpc-google-cloud-pubsub-v1:jar:0.1.18 -> io.grpc:grpc-protobuf:jar:1.6.1 -> com.google.protobuf:protobuf-java:jar:3.3.1
	at com.google.cloud.tools.opensource.dependencies.DependencyGraphIntegrationTest.testFindUpdates_beam(DependencyGraphIntegrationTest.java:85)
Caused by: org.eclipse.aether.resolution.ArtifactDescriptorException: Failed to read artifact descriptor for com.google.protobuf:protobuf-java:jar:3.3.1
	at com.google.cloud.tools.opensource.dependencies.DependencyGraphIntegrationTest.testFindUpdates_beam(DependencyGraphIntegrationTest.java:85)
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not transfer artifact com.google.protobuf:protobuf-java:pom:3.3.1 from/to central (http://repo1.maven.org/maven2/): Connect to repo1.maven.org:80 timed out
	at com.google.cloud.tools.opensource.dependencies.DependencyGraphIntegrationTest.testFindUpdates_beam(DependencyGraphIntegrationTest.java:85)
Caused by: org.eclipse.aether.transfer.ArtifactTransferException: Could not transfer artifact com.google.protobuf:protobuf-java:pom:3.3.1 from/to central (http://repo1.maven.org/maven2/): Connect to repo1.maven.org:80 timed out
	at com.google.cloud.tools.opensource.dependencies.DependencyGraphIntegrationTest.testFindUpdates_beam(DependencyGraphIntegrationTest.java:85)
Caused by: org.apache.http.conn.ConnectTimeoutException: Connect to repo1.maven.org:80 timed out
	at com.google.cloud.tools.opensource.dependencies.DependencyGraphIntegrationTest.testFindUpdates_beam(DependencyGraphIntegrationTest.java:85)

```

Will try later.

File: dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtility.java [hunk 1]
```diff
@@ -86,6 +87,12 @@
   
   public static final RemoteRepository CENTRAL =
       new RemoteRepository.Builder("central", "default", "http://repo1.maven.org/maven2/").build();
+  public static final RemoteRepository GOOGLE =
```
Comment (suztomo) [inline dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtility.java]: Probably this should be configured at setting.xml of Kokoro user. But I wanted to see how this helps the timeout issue for now.

File: dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtility.java [hunk 2]
```diff
@@ -86,8 +87,15 @@
...
+      new RemoteRepository.Builder(
+              "google",
+              "default",
+              "https://maven-central.storage-download.googleapis.com/repos/central/data/")
+          .build();
 
-  private static ImmutableList<RemoteRepository> mavenRepositories = ImmutableList.of(CENTRAL);
+  private static ImmutableList<RemoteRepository> mavenRepositories =
+      ImmutableList.of(GOOGLE, CENTRAL);
```
Comment (suztomo) [inline dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtility.java]: Maybe this should be an option.

File: dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtility.java [hunk 3]
```diff
@@ -84,10 +84,11 @@
 
   private static final Logger logger = Logger.getLogger(RepositoryUtility.class.getName());
   
-  public static final RemoteRepository CENTRAL =
-      new RemoteRepository.Builder("central", "default", "http://repo1.maven.org/maven2/").build();
+  public static final RemoteRepository GOOGLE_MIRROR =
```
Comment (elharo) [inline dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtility.java]: Adding the Google mirror is fine, but I'm not sure we want to remove the central repository.

Comment (suztomo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtility.java]: This change is not to be merged. I'm experimenting how it solves (or not).

**Comment by suztomo** [comment]
Replacing CENTRAL with Google's mirror worked without timeout. Trying adding google mirror in addition to central.

File: kokoro/continuous.bat [hunk 4]
```diff
@@ -4,6 +4,10 @@ set JAVA_HOME=c:\program files\java\jdk1.8.0_152
 set PATH=%JAVA_HOME%\bin;%PATH%
 
 cd github/cloud-opensource-java
+
+mkdir %USERPROFILE%\.m2
```
Comment (suztomo) [inline kokoro/continuous.bat]: With "mvn -X", I confirmed this matches user.home in Kokoro GCP_WINDOWS.

**Comment by suztomo** [comment]
@elharo PTAL. Now ready for review.

**Comment by suztomo** [comment]
Sure. let me fix.

**Comment by suztomo** [comment]
@elharo PTAL. Now change is only LinkageCheckerArgumentsTest..

**Pull request merged**